### ISSUE TITLE
Use explicit VAO in all examples

### ISSUE THREAD

**Issue by pjcozzi** [WebGLSamples/WebGL2Samples#16]
WebGL 2 has a default Vertex Array Object that we are implicitly using in some of the examples.

I think it would be a better practice if we use an explicit VAO like the `sampler_wrap` example:

``` javascript
            var vertexArray = gl.createVertexArray();
            gl.bindVertexArray(vertexArray);

            gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer);
            gl.vertexAttribPointer(positionLocation, 2, gl.FLOAT, false, 0, 0);
            gl.enableVertexAttribArray(positionLocation);
            gl.bindBuffer(gl.ARRAY_BUFFER, null);

            gl.bindBuffer(gl.ARRAY_BUFFER, textureBuffer);
            gl.vertexAttribPointer(texcoordLocation, 2, gl.FLOAT, false, 0, 0);
            gl.enableVertexAttribArray(texcoordLocation);
            gl.bindBuffer(gl.ARRAY_BUFFER, null);

            gl.bindVertexArray(null);
```

Then when rendering:

``` javascript
gl.bindVertexArray(vertexArray);
gl.drawArrays(/* ... */);
```

**Comment by shrekshao** [comment]
Done for samples on master

**Comment by pjcozzi** [comment]
Fixed in 92bc36d43822280d4f7f166e0a1fef2b5f9be347.

Thanks @shrekshao.  In the future, instead of committing to master, make changes on a new branch, and open a pull request.
